zfs/zfs.go

```diff
@@ -1006,6 +1034,40 @@ func (e *ErrRecvResumeNotSupported) Error() string {
 	return buf.String()
 }
 
+type RecvOptions struct {
+	// Rollback to the oldest snapshot, destroy it, then perform `recv -F`.
+	// Note that this doesn't change property values, i.e. an existing local property value will be kept.
+	RollbackAndForceRecv bool
+	// Set -s flag used for resumable send & recv
+	SavePartialRecvState bool
+
+	InheritProperties  []zfsprop.Property
+	OverrideProperties map[zfsprop.Property]string
+}
+
+func (opts RecvOptions) buildCommonRecvArgs() []string {
```

Comment:
Why is this method called `buildCommonRecvArgs` and not just `buildRecvArgs`? The method is used in the same way from all call sites, the call sites don't extend the returned argv with more flags

Reply:
Funnily enough I named it so because I borrowed the naming from buildCommonSendArgs, which is used as you describe and includes specific things like snapshots and dataset names, while buildCommonRecvArgs actually does only build the 'common' args, as in flags from booleans.
For some of the testing I added it would be nice to split SendArgsUnvalidated.buildCommonSendArgs() into SendArgsValidated.buildSendArgs() (as suggested), which gets its flags from SendArgsUnvalidated.buildCommonSendArgs().
Not sure if this is beneficial beyond being able to test it.